### ISSUE TITLE
Not able to connect via Bluetooth

### ISSUE THREAD

**Issue by emiliomarin** [Parrot-Developers/RollingSpiderEdu#50]
Hi, I'm following the steps from the Getting Started file. I am using a virtual machine and a Bluetooth 4.0 adapter. I'm having trouble flashing the drone. When I do DroneConnect.sh I get this as a result

```
> Drone: Trying to read drone MAC address from file...
> Drone: Attempting connection with address: A0:14:3D:84:43:A3...
pand[3020]: Bluetooth PAN daemon version 4.101
pand[3020]: Connecting to A0:14:3D:84:43:A3
pand[3020]: Connect to A0:14:3D:84:43:A3 failed. Host is down(112)
SIOCSIFADDR: No such device
bnep0: ERROR while getting interface flags: No such device
bnep0: ERROR while getting interface flags: No such device
> Drone: ERROR: Bluetooth connection could was not established.
```

If I go to my bluetooth settings I can see the drone : RS_B03... but if I try to connect from there it says: "Setting up RS_B03... failed"

Also, I tried to do sudo hciconfig lescan and as a result I get this:

```
E0:14:43:A3:3D:84 RS_B038406
E0:14:43:A3:3D:84 (unknown)
```

That address is different to the one I found in the drone and used in DroneSetMACaddress.sh. Not sure if it has something to do, I tried changing the MAC to that one but I get the same result.

Any thoughts?

**Comment by nbibin001** [comment]
Hey I have had a lot of trouble with virtual machine connecting to drone via Bluetooth. Finally I decided to install Debian full OS , alongside my windows operating system. And after that when I tried the connection to the drone, it just worked magically. I would really suggest trying it on a full Debian OS, there are descriptive tutorials available for this online.

**Comment by emiliomarin** [comment]
Ok, I will give that a try!

**Comment by emiliomarin** [comment]
@nbibin001 Thanks for your answer before, I installed Ubuntu in a partition in my hard drive. Now I am getting this, it says that it's connected but doesn't upload anything. If I do DroneDisconnect.sh and then DroneConnect.sh I get same result.

```
emilio@emilio:~$ DroneConnect.sh 
Drone: Drone seems to be connected. You might want to try a DroneDisconnect.sh first.
emilio@emilio:~$ EDUfirmwareUploadFILES.sh 
Drone: Trying to upload custom firmware files...
> Drone: FTP into the drone
spawn lftp -e mirror -R ../../libs/EDUfirmwareFILES/ / -u anonymous, 192.168.1.1
mkdir `/' [Connecting...]
```

**Comment by nbibin001** [comment]
Hi emiliomarin , I remember running into an issue like this too. I guess what might be happening is if your ubuntu linux computer is connected to wifi router, which most of the cases will have an IP address of 192.168.1.1. So the Droneconnect.sh thinks that it is already connected to the drone, because drone and all these shell scripts use the same ip address 192.168.1.1 for communication. So I had to disable wifi networks and then tried DroneDisconnect and DroneConnect, it worked for me. You might want to do a restart and after disabling wifi I guess. There is option to disable wifi on the top right corner.

**Comment by ghost** [comment]
Hi,
I am also encountering a similar issue. I am using Debian OS. Whenever I am trying to execute the DroneConnect.sh I am receiving the following error
> Drone: Trying to read drone MAC address from file...
> Drone: Attempting connection with address: A0:14:3D:4B:10:5F...
pand[9187]: Bluetooth PAN daemon version 4.99
pand[9187]: Connecting to A0:14:3D:4B:10:5F
pand[9187]: Connect to A0:14:3D:4B:10:5F failed. Connection refused(111)
SIOCSIFADDR: No such device
bnep0: ERROR while getting interface flags: No such device
bnep0: ERROR while getting interface flags: No such device
> Drone: ERROR: Bluetooth connection could was not established.

Do I need to migrate a Ubuntu workstation to resolve this issue?

**Comment by nbibin001** [comment]
Are you using the Ubuntu virtual machine ?

**Comment by ghost** [comment]
No, I am using Debian OS. Is Ubuntu mandatory?

**Comment by cmeissen** [comment]
Ankur, were you able to solve this problem? I get the same error using an Ubuntu 14.04 VM. Thanks

**Comment by ghost** [comment]
Yes,
I solved the problem both in Ubuntu 14.04 and Debian OS. My problem was solely because I did not have a bluetooth adapter supported on Linux and also that supports BLE.
My suggestion for you would be to check if your Bluetooth is BLE and supports Linux.

**Comment by vlngod** [comment]
https://www.youtube.com/watch?v=CeW5PE8akaM 
************Fix bluetooth connectivity issue*******************
# open a terminal and run the following commands:
1. sudo apt-get install libspa-0.2-modules=0.3.19-4
2. sudo apt install libspa-0.2-bluetooth && apt purge pulseaudio-module-bluetooth

------Now reboot your system: 
       sudo reboot now
